### PR TITLE
select correct entity for tile, and fix isOccupied and combat

### PR DESCRIPTION
**Pull request by lallenlowe**

### CODE REVIEW

File: src/map.js [hunk 1]
```diff
@@ -59,13 +59,15 @@ class Map {
...
+		return entity !== undefined && entity.isAlive();
 	}
 
 	getEntityAt(x, y) {
-		let entity = this.getEntities();
+		let entities = this.getEntities();
 
-		return entity[this.makeKey(x, y)];
+		return _.find(entities, (entity) => {
+			return entity.getX() == x && entity.getY() == y;
```
Comment (inphomercial) [inline src/map.js]: Is this returning a `bool`?

File: src/map.js [hunk 2]
```diff
@@ -59,13 +59,15 @@ class Map {
 	isTileOccupied(x, y) {
 		let entity = this.getEntityAt(x, y);
 
-		return entity !== undefined;
+		return entity !== undefined && entity.isAlive();
```
Comment (inphomercial) [inline src/map.js]: We may not want to tie in an `isAlive` here, but is that part of the problem we were facing?

File: src/map.js (hunk 1 above)
Comment (inphomercial) [reply src/map.js]: Oh i see how `._find` works.. nice job~

Comment (lallenlowe) [reply src/map.js]: yes, maybe we should use typescript?? lol

**Pull request merged**